server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -407,6 +407,7 @@ void recoverFromSnapshotAndRemoteStore(
                     threadPool
                 );
                 RemoteSegmentStoreDirectory sourceRemoteDirectory = (RemoteSegmentStoreDirectory) directoryFactory.newDirectory(
+                    remoteStoreRepository,
```

Comment:
so, right now we pull this remote store repository value from shard level snapshot metadata, we need to make related changes there to update that with metadata directory now.

Reply:
Also, i don't like the idea of using same repository name here for metadata and data, we should maybe create a remote directory instance from directoryFactory which have access to metadata repository only.

Reply:
checked more on this, actually we will need segments_N file for restoring snapshot, which is in data directory, so we will need to add both segment data and metadata repository info in snapshot metadata. unless we keep segments_N file in metadata directory as well.